meson.build

```diff
@@ -1,5 +1,5 @@
 project('harfbuzz', 'c', 'cpp',
-  meson_version: '>= 0.55.0',
+  meson_version: '>= 1.3.0',
```

Comment:
This is a pretty major jump. It might be justifiable (and I haven't looked into specifics) but I do recall a while back (~1.5 years?) there was some blocker downstream that needed old Meson support. Perhaps that was an Ubuntu LTS release? I don't actually remember, but just want to check that we're sure moving to 1.3.0 isn't going to leave any distros that switched from autotools to meson out in the cold.